inner_products/src/lib.rs

```diff
@@ -68,22 +68,18 @@ impl<P: PairingEngine> InnerProduct for PairingInnerProduct<P> {
                 right.len(),
             )));
         };
-        let aff_left = P::G1Projective::batch_normalization_into_affine(left);
-        let aff_right = P::G2Projective::batch_normalization_into_affine(right);
-        let aff_pairs = cfg_into_iter!(aff_left)
-            .zip(aff_right)
-            .map(|(a, b)| (P::G1Prepared::from(a), P::G2Prepared::from(b)))
-            .collect::<Vec<_>>();
-        Ok(ExtensionFieldElement(P::product_of_pairings(&aff_pairs)))
+        let aff_left = P::G1::normalize_batch(left);
+        let aff_right = P::G2::normalize_batch(right);
+        Ok(P::multi_pairing(&aff_left, &aff_right))
```

Comment:
Ditto here: might be worth it manually chunk up `aff_left` and `aff_right` and perform a multi-pairing on the chunks.

Reply:
Might be worth making a PR to algebra with the parallelized multi-pairing, but in the meanwhile we can write a helper function here.

Reply:
I exposed a `cfg_multi_pairing` function that does this